"cargo install cargo-semver-checks" fails for me at the moment

### Steps to reproduce the bug with the above code

On an M1 Mac with Rust 1.65, running `cargo install cargo-semver-checks` hits a build error:

```
176 |                 Ok(VersionedRustdocAdapter::V22(
    |                    ---------------------------- arguments to this enum variant are incorrect
177 |                     trustfall_rustdoc_adapter_v22::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
```

### Actual Behaviour

```
% cargo install cargo-semver-checks

    Updating crates.io index
  Downloaded cargo-semver-checks v0.13.1
  Downloaded 1 crate (50.5 KB) in 0.18s
  Installing cargo-semver-checks v0.13.1
  Downloaded pest_derive v2.5.1
  Downloaded clap-verbosity-flag v2.0.0
  Downloaded is-terminal v0.4.1
  Downloaded io-lifetimes v1.0.3
  Downloaded handlebars v4.3.5
  Downloaded pest_generator v2.5.1
  Downloaded rustdoc-types v0.12.0
  Downloaded sys-info v0.9.1
  Downloaded ucd-trie v0.1.5
  Downloaded trustfall_filetests_macros v0.2.0
  Downloaded rustdoc-types v0.17.0
  Downloaded ron v0.7.1
  Downloaded trustfall_core v0.0.7
  Downloaded cargo_metadata v0.15.2
  Downloaded chrono v0.4.23
  Downloaded git-version-macro v0.3.5
  Downloaded clap v4.0.29
  Downloaded typenum v1.16.0
  Downloaded sha1 v0.10.5
  Downloaded git-version v0.3.5
  Downloaded trustfall_rustdoc v0.5.0
  Downloaded rayon-core v1.10.1
  Downloaded termcolor_output_impl v1.0.0
  Downloaded rustix v0.36.5
  Downloaded trustfall_core v0.1.1
  Downloaded termcolor_output v1.0.1
  Downloaded rayon v1.6.0
  Downloaded trustfall-rustdoc-adapter v22.3.0
  Downloaded async-graphql-parser v2.11.3
  Downloaded pest_meta v2.5.1
  Downloaded trustfall-rustdoc-adapter v16.3.0
  Downloaded trustfall-rustdoc-adapter v21.3.0
  Downloaded human-panic v1.0.3
  Downloaded crates-index v0.18.11
  Downloaded rustdoc-types v0.18.0
  Downloaded iana-time-zone v0.1.53
  Downloaded openssl-sys v0.9.79
  Downloaded bugreport v0.5.0
  Downloaded async-graphql-value v2.11.3
  Downloaded clap-cargo v0.10.0
  Downloaded cargo_toml v0.13.0
  Downloaded ron v0.6.6
  Downloaded pest v2.5.1
  Downloaded os_type v2.6.0
  Downloaded 44 crates (3.1 MB) in 0.75s
   Compiling proc-macro2 v1.0.47
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.5
   Compiling syn v1.0.105
   Compiling autocfg v1.1.0
   Compiling serde_derive v1.0.149
   Compiling libc v0.2.138
   Compiling serde v1.0.149
   Compiling memchr v2.5.0
   Compiling thiserror v1.0.37
   Compiling log v0.4.17
   Compiling cfg-if v1.0.0
   Compiling bitflags v1.3.2
   Compiling serde_json v1.0.89
   Compiling version_check v0.9.4
   Compiling ryu v1.0.11
   Compiling itoa v1.0.4
   Compiling ucd-trie v0.1.5
   Compiling pkg-config v0.3.26
   Compiling regex-syntax v0.6.28
   Compiling once_cell v1.16.0
   Compiling indexmap v1.9.2
   Compiling aho-corasick v0.7.20
   Compiling jobserver v0.1.25
   Compiling num-traits v0.2.15
   Compiling camino v1.1.1
   Compiling core-foundation-sys v0.8.3
   Compiling cc v1.0.77
   Compiling num-integer v0.1.45
   Compiling hashbrown v0.12.3
   Compiling either v1.8.0
   Compiling crossbeam-utils v0.8.14
   Compiling bstr v0.2.17
   Compiling fnv v1.0.7
   Compiling same-file v1.0.6
   Compiling base64 v0.13.1
   Compiling walkdir v2.3.2
   Compiling regex v1.7.0
   Compiling iana-time-zone v0.1.53
   Compiling time v0.1.45
   Compiling proc-macro-error-attr v1.0.4
   Compiling memoffset v0.7.1
   Compiling lazy_static v1.4.0
   Compiling io-lifetimes v1.0.3
   Compiling globset v0.4.9
   Compiling libz-sys v1.1.8
   Compiling openssl-sys v0.9.79
   Compiling itertools v0.10.5
   Compiling proc-macro-error v1.0.4
   Compiling crossbeam-epoch v0.9.13
   Compiling maplit v1.0.2
   Compiling rustix v0.36.5
   Compiling tinyvec_macros v0.1.0
   Compiling tinyvec v1.6.0
   Compiling errno v0.2.8
   Compiling termcolor v1.1.3
   Compiling semver v1.0.14
   Compiling scopeguard v1.1.0
   Compiling proc-macro-hack v0.5.19
   Compiling thiserror-impl v1.0.37
   Compiling trustfall_filetests_macros v0.2.0
   Compiling unicode-normalization v0.1.22
   Compiling sys-info v0.9.1
   Compiling libgit2-sys v0.14.0+1.5.0
   Compiling unicode-bidi v0.3.8
   Compiling percent-encoding v2.2.0
   Compiling os_str_bytes v6.4.1
   Compiling heck v0.4.0
   Compiling rayon-core v1.10.1
   Compiling clap_lex v0.3.0
   Compiling form_urlencoded v1.1.0
   Compiling clap_derive v4.0.21
   Compiling pest v2.5.1
   Compiling idna v0.3.0
   Compiling is-terminal v0.4.1
   Compiling crossbeam-deque v0.8.2
   Compiling pest_meta v2.5.1
   Compiling crossbeam-channel v0.5.6
   Compiling backtrace v0.3.66
   Compiling num_cpus v1.14.0
   Compiling smartstring v1.0.1
   Compiling pest_generator v2.5.1
   Compiling doc-comment v0.3.3
   Compiling adler v1.0.2
   Compiling gimli v0.26.2
   Compiling anyhow v1.0.66
   Compiling strsim v0.10.0
   Compiling clap v4.0.29
   Compiling miniz_oxide v0.5.4
   Compiling pest_derive v2.5.1
   Compiling git-version-macro v0.3.5
   Compiling url v2.3.1
   Compiling getrandom v0.2.8
   Compiling object v0.29.0
   Compiling rustc-demangle v0.1.21
   Compiling addr2line v0.17.0
   Compiling bugreport v0.5.0
   Compiling static_assertions v1.1.0
   Compiling git-version v0.3.5
   Compiling bytes v1.3.0
   Compiling chrono v0.4.23
   Compiling ron v0.6.6
   Compiling smallvec v1.10.0
   Compiling async-graphql-value v2.11.3
   Compiling toml v0.5.9
   Compiling async-graphql-parser v2.11.3
   Compiling rustdoc-types v0.17.0
   Compiling trustfall_core v0.1.1
   Compiling rustdoc-types v0.18.0
   Compiling cargo-platform v0.1.2
   Compiling rustdoc-types v0.12.0
   Compiling cargo_metadata v0.15.2
   Compiling trustfall_core v0.0.7
   Compiling hex v0.4.3
   Compiling rayon v1.6.0
   Compiling uuid v0.8.2
   Compiling trustfall-rustdoc-adapter v16.3.0
   Compiling trustfall-rustdoc-adapter v22.3.0
   Compiling trustfall-rustdoc-adapter v21.3.0
   Compiling thread_local v1.1.4
   Compiling os_type v2.6.0
   Compiling termcolor_output_impl v1.0.0
   Compiling rustc-hash v1.1.0
   Compiling shell-escape v0.1.5
   Compiling home v0.5.4
   Compiling termcolor_output v1.0.1
   Compiling human-panic v1.0.3
   Compiling ignore v0.4.18
   Compiling clap-cargo v0.10.0
   Compiling trustfall_rustdoc v0.5.0
error[E0277]: the trait bound `trustfall_rustdoc_adapter::RustdocAdapter<'_>: BasicAdapter<'_>` is not satisfied
  --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/query.rs:78:30
   |
78 |                 interpret_ir(adapter.clone(), parsed_query, parsed_vars).map_err(|e| e.into())
   |                 ------------ ^^^^^^^^^^^^^^^ the trait `BasicAdapter<'_>` is not implemented for `trustfall_rustdoc_adapter::RustdocAdapter<'_>`
   |                 |
   |                 required by a bound introduced by this call
   |
   = help: the following other types implement trait `Adapter<'token>`:
             <AdapterTap<'token, DataToken, AdapterT> as Adapter<'token>>
             <trustfall_core::filesystem_interpreter::FilesystemInterpreter as Adapter<'static>>
   = note: required for `trustfall_rustdoc_adapter::RustdocAdapter<'_>` to implement `Adapter<'_>`
note: required by a bound in `interpret_ir`
  --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_core-0.0.7/src/interpreter/execution.rs:32:30
   |
32 |     adapter: Rc<RefCell<impl Adapter<'query, DataToken = DataToken> + 'query>>,
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `interpret_ir`

error[E0277]: the trait bound `trustfall_rustdoc_adapter::RustdocAdapter<'_>: BasicAdapter<'_>` is not satisfied
  --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/query.rs:83:30
   |
83 |                 interpret_ir(adapter.clone(), parsed_query, parsed_vars).map_err(|e| e.into())
   |                 ------------ ^^^^^^^^^^^^^^^ the trait `BasicAdapter<'_>` is not implemented for `trustfall_rustdoc_adapter::RustdocAdapter<'_>`
   |                 |
   |                 required by a bound introduced by this call
   |
   = help: the following other types implement trait `Adapter<'token>`:
             <AdapterTap<'token, DataToken, AdapterT> as Adapter<'token>>
             <trustfall_core::filesystem_interpreter::FilesystemInterpreter as Adapter<'static>>
   = note: required for `trustfall_rustdoc_adapter::RustdocAdapter<'_>` to implement `Adapter<'_>`
note: required by a bound in `interpret_ir`
  --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_core-0.0.7/src/interpreter/execution.rs:32:30
   |
32 |     adapter: Rc<RefCell<impl Adapter<'query, DataToken = DataToken> + 'query>>,
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `interpret_ir`

error[E0277]: the trait bound `trustfall_rustdoc_adapter::RustdocAdapter<'_>: BasicAdapter<'_>` is not satisfied
  --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/query.rs:88:30
   |
88 |                 interpret_ir(adapter.clone(), parsed_query, parsed_vars).map_err(|e| e.into())
   |                 ------------ ^^^^^^^^^^^^^^^ the trait `BasicAdapter<'_>` is not implemented for `trustfall_rustdoc_adapter::RustdocAdapter<'_>`
   |                 |
   |                 required by a bound introduced by this call
   |
   = help: the following other types implement trait `Adapter<'token>`:
             <AdapterTap<'token, DataToken, AdapterT> as Adapter<'token>>
             <trustfall_core::filesystem_interpreter::FilesystemInterpreter as Adapter<'static>>
   = note: required for `trustfall_rustdoc_adapter::RustdocAdapter<'_>` to implement `Adapter<'_>`
note: required by a bound in `interpret_ir`
  --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_core-0.0.7/src/interpreter/execution.rs:32:30
   |
32 |     adapter: Rc<RefCell<impl Adapter<'query, DataToken = DataToken> + 'query>>,
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `interpret_ir`

error[E0308]: mismatched types
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:123:21
    |
122 |                 Ok(VersionedRustdocAdapter::V16(
    |                    ---------------------------- arguments to this enum variant are incorrect
123 |                     trustfall_rustdoc_adapter_v16::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
note: tuple variant defined here
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:52:5
    |
52  |     V16(
    |     ^^^

   Compiling cargo_toml v0.13.0
error[E0308]: mismatched types
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:133:21
    |
132 |                 Ok(VersionedRustdocAdapter::V16(
    |                    ---------------------------- arguments to this enum variant are incorrect
133 |                     trustfall_rustdoc_adapter_v16::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
note: tuple variant defined here
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:52:5
    |
52  |     V16(
    |     ^^^

   Compiling handlebars v4.3.5
error[E0308]: mismatched types
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:145:21
    |
144 |                 Ok(VersionedRustdocAdapter::V21(
    |                    ---------------------------- arguments to this enum variant are incorrect
145 |                     trustfall_rustdoc_adapter_v21::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
note: tuple variant defined here
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:58:5
    |
58  |     V21(
    |     ^^^

   Compiling ron v0.7.1
   Compiling clap-verbosity-flag v2.0.0
error[E0308]: mismatched types
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:155:21
    |
154 |                 Ok(VersionedRustdocAdapter::V21(
    |                    ---------------------------- arguments to this enum variant are incorrect
155 |                     trustfall_rustdoc_adapter_v21::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
note: tuple variant defined here
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:58:5
    |
58  |     V21(
    |     ^^^

error[E0308]: mismatched types
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:167:21
    |
166 |                 Ok(VersionedRustdocAdapter::V22(
    |                    ---------------------------- arguments to this enum variant are incorrect
167 |                     trustfall_rustdoc_adapter_v22::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
note: tuple variant defined here
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:64:5
    |
64  |     V22(
    |     ^^^

error[E0308]: mismatched types
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:177:21
    |
176 |                 Ok(VersionedRustdocAdapter::V22(
    |                    ---------------------------- arguments to this enum variant are incorrect
177 |                     trustfall_rustdoc_adapter_v22::RustdocAdapter::schema(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `Schema`, found struct `trustfall_core::schema::Schema`
    |
    = note: perhaps two different versions of crate `trustfall_core` are being used?
note: tuple variant defined here
   --> /Users/mstange/.cargo/registry/src/github.com-1ecc6299db9ec823/trustfall_rustdoc-0.5.0/src/versioned.rs:64:5
    |
64  |     V22(
    |     ^^^

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `trustfall_rustdoc` due to 9 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-semver-checks v0.13.1`, intermediate artifacts can be found at `/var/folders/n0/ks1j6rkx4zg1q4kfd06cmt9r0000gn/T/cargo-install16GU0b`
```

### Expected Behaviour

Installation should work.

### Generated System Information

```
% rustup show

Default host: aarch64-apple-darwin
rustup home:  /Users/mstange/.rustup

[...]

active toolchain
----------------

stable-aarch64-apple-darwin (default)
rustc 1.65.0 (897e37553 2022-11-02)
```

### Additional Context

_No response_

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "cargo-semver-checks"
-version = "0.13.1"
+version = "0.14.0"
 edition = "2021"
 authors = ["Predrag Gruevski <obi1kenobi82@gmail.com>"]
 license = "Apache-2.0 OR MIT"

